tutorial-review.md

```diff
@@ -0,0 +1,1053 @@
+
+const methods = {
+  get,
+  post,
+  put,
+  delete: deleteData,
+};
+
+export default methods;
+```
+
+We will see how to use the `http` object in the next section.
+
+### Create containers and reusable components
+
+The very first component we are going to build is `Home`, in charge of rendering the list of posts as well as the header of the homepage.
+To render the list of posts Home has to 
+1. Call the server GET `/api/posts` endpoint after the first rendering
+2. Store the array posts in the state
+3. Render the posts to the user and link them to `/posts/:post_id` to read the content
+
+In the terminal create a folder `/pages` and a file `home.js` in it:
+
+```bash
+mkdir pages
+touch pages/home.js
+```
+Add the following code in it:
+
+```javascript
+import { useEffect, useState } from 'react';
+// Link component allow users to navigate to the blog post component page
+import { Link } from 'react-router-dom';
+import Container from 'react-bootstrap/Container';
+import ListGroup from 'react-bootstrap/ListGroup';
+import Image from 'react-bootstrap/Image';
+import http from '../lib/http';
+// utility function to format the creation date
+import formatDate from '../lib/formatDate';
+
+const Home = () => {
+  // useState allows us to make use of the component state to store the posts
+  const [posts, setPosts] = useState([]); 
+  useEffect(() => {
+    // Call the server to fetch the posts and store them into the state
+    async function fetchData() {
+      const { data } = await http.get('/api/posts');
+      setPosts(data.data.posts);
+    }
+    fetchData();
+  }, []);
+  
+  return (
+    <>
+      <Container className="my-5" style={{ maxWidth: '800px' }}>
+        <Image
+          src="avatar.jpeg"
+          width="150"
+          style={{ borderRadius: '50%' }}
+          className="d-block mx-auto img-fluid"
+        />
+        <h2 className="text-center">Welcome to the Digital Marketing blog</h2>
+      </Container>
+      <Container style={{ maxWidth: '800px' }}>
+        <ListGroup variant="flush" as="ol">
+          {
+            posts.map((post) => {
+              // Map the posts to JSX
+              return (
+                <ListGroup.Item key={post._id}> 
+                  <div className="fw-bold h3">
+                    <Link to={`/posts/${post._id}`} style={{ textDecoration: 'none' }}>{post.title}</Link>
+                  </div>
+                  <div>{post.author} - <span className="text-secondary">{formatDate(post.createdAt)}</span></div>
+                </ListGroup.Item>
+              );
+            })
+          }
+        </ListGroup>
+      </Container>
+    </>
+  );
+};
+
+export default Home;
+```
+
+One special mention goes to `formatDate`, a utility function that formats the post creation date to "MMMM DD, YYYY". We are expecing to call it in other components as well, this is why it is decoupled from `Home` into its own file.
```

Comment:
For the post creation date, is it MMMM or MM?

Reply:
I wrote MMMM as the month is gonna be displayed as a string "January", not "01".